Bexchange/Controllers/BookController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using AutoMapper;
+using Bexchange.Infrastructure.Repositories.Interfaces;
+using BexchangeAPI.Domain.CustomExceptions;
+using BexchangeAPI.Domain.Enum;
+using BexchangeAPI.Domain.Models;
+using BexchangeAPI.DTOs;
+using BexchangeAPI.Infrastructure.Repositories.Interfaces;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using System.Net;
+using System.Security.Claims;
+
+namespace BexchangeAPI.Controllers
+{
+    [Authorize]
+    [Route("api/[controller]")]
+    [ApiController]
+    public class BookController : ControllerBase
+    {
+        private readonly IContentRepository<Book> _contentRepo;
+        public readonly IUsersRepository<User> _usersRepository;
+        private readonly IMapper _mapper;
+
+        public BookController(IContentRepository<Book> contentRepo, IUsersRepository<User> usersRepository, IMapper mapper)
+        {
+            _mapper = mapper;
+            _contentRepo = contentRepo;
+            _usersRepository = usersRepository;
+        }
+
+        [HttpGet, AllowAnonymous]
+        public async Task<IActionResult> Books()
+        {
+            var books = await _contentRepo.GetAllComponentsAsync();
+
+            if (books == null)
+                throw new NotFoundException("No books here", (int)HttpStatusCode.NotFound);
+
+            return Ok(_mapper.Map<IEnumerable<BookDto>>(books));
+
+        }
+
+        [HttpGet("{id}")]
+        public async Task<IActionResult> GetBook(int id)
+        {
+            var book = await _contentRepo.GetComponentAsync(id);
+
+            if (book == null)
+                throw new NotFoundException("Book not found", (int)HttpStatusCode.NotFound);
+
+            return Ok(_mapper.Map<BookDto>(book));
+        }
+
+        [HttpGet("user/{userId}")]
+        public async Task<IActionResult> GetUserBooks(int userId)
+        {
+            var books = await _contentRepo.GetUserComponentsAsync(userId);
+
+            if (books == null)
+                throw new NotFoundException("Books not found", (int)HttpStatusCode.NotFound);
+
+            return Ok(_mapper.Map<IEnumerable<BookDto>>(books));
+        }
+
+        [HttpPost("add")]
+        public async Task<IActionResult> AddBook(BookDto book)
+        {
+            var newBook = _mapper.Map<Book>(book);
+
+            newBook.UserId = GetUserId();
+            newBook.User = await _usersRepository.GetUserAsync(newBook.UserId);
+            await _contentRepo.AddComponentAsync(newBook);
+
+            return Created(Request.Path, new { newBook.Id });
```

Comment:
Created should return the link to element which was created. Are you sure that link is valid?